pages/search.tsx

```diff
@@ -0,0 +1,238 @@
+    margin-right: 0.5rem;
+  }
+`
+
+const FilterSection = styled(Section)`
+  &:not(:last-child) {
+    margin-bottom: 1.25rem;
+  }
+`
+
+const FilterSectionHeadline = styled(Typography).attrs(() => ({ variant: "body_short_bold" }))`
+  margin-bottom: 0.5rem;
+`
+
+const TagsContainer = styled.div`
+  display: flex;
+  flex-wrap: wrap;
+  grid-gap: 0.5rem;
+`
+
+const Tag = styled(Button)<{ active: boolean }>`
+  padding: 0 0.5rem;
+  font-size: 0.75rem;
+  background-color: ${tokens.colors.infographic.primary__moss_green_13.rgba};
+  border: none;
+  color: ${tokens.colors.text.static_icons__default.hex};
+  ${({ active }) => active && css`
+    box-shadow: 0 0 0 2px ${tokens.colors.infographic.primary__moss_green_55.rgba};
+  `}
+
+  &:hover {
+    background-color: ${tokens.colors.infographic.primary__moss_green_34.rgba};
+    border: none;
+  }
+`
+
+const SearchResultsList = styled.ul`
+  padding: 0;
+  margin: 0;
+  list-style: none;
+`
+
+const Search: NextPage = () => {
+  const [isLoading, setIsLoading] = useState<boolean>(true)
+  const [communities, setCommunities] = useState<any[]>([])
+  const [searchResults, setSearchResults] = useState<any[]>([])
+
+  const router = useRouter()
+
+  useEffect(() => {
+    (async () => {
+      try {
+        const { body } = await HttpClient.get("/api/collibra/communities", {
+          headers: { authorization: `Bearer ${localStorage.getItem("access_token")}` },
+          query: {
+            // TODO: get this ID from Collibra
+            parentId: "1f63fe17-a732-4779-a290-db088803761c",
+          },
+        })
+        setCommunities(body.results)
+      } catch (error) {
+        console.error("[Search] Failed fetching communities", error)
+      }
+    })()
+  }, [])
+
+  useEffect(() => {
+    setIsLoading(true)
+
+    if (router.query.q) {
+      (async () => {
+        try {
+          const { body } = await HttpClient.get("/api/search", {
+            headers: { authorization: `Bearer ${localStorage.getItem("access_token")}` },
+            query: router.query,
+          })
+
+          setSearchResults(body.results.map((result: any) => result.resource))
+          setIsLoading(false)
+        } catch (error) {
+          console.error("[Search] Failed fetching search results", error)
+          setIsLoading(false)
+        }
+      })()
+    }
+  }, [router])
+
+  const onCommunityFilterClick = (id: string) => {
+    let { community } = router.query
+
+    if (!community) {
+      community = id
+    } else if (typeof community === "string") {
+      if (community === id) {
+        community = undefined
+      } else {
+        community = [community, id]
+      }
+    } else {
+      const existingIdx = community.findIndex((f) => f === id)
+      if (existingIdx === -1) {
+        community.push(id)
+      } else {
+        community = [
+          ...community.slice(0, existingIdx),
+          ...community.slice(existingIdx + 1, community.length),
+        ]
+      }
```

Comment:
An alternative approach since we only use the index for existence check:
```
      const filteredCommunity = community.filter((f) => f !== id)
      if (filteredCommunity.length === community.length) {
         community.push(id)
      } else {
        community = filteredCommunity
      }
```

Reply:
we also use the index to be able to remove an item at the given index. if a user has clicked a community in the filter list, they should be able to remove it by clicking it again.

Reply:
Yup, the suggested codeblock also covers that without needing the index.
Nice work with the typedoc and tests 👏🏻